homeassistant/components/binary_sensor/digital_ocean.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for monitoring the state of Digital Ocean droplets.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.digital_ocean/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import CONF_NAME
+from homeassistant.loader import get_component
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CREATED_AT = 'created_at'
+ATTR_DROPLET_ID = 'droplet_id'
+ATTR_DROPLET_NAME = 'droplet_name'
+ATTR_FEATURES = 'features'
+ATTR_IPV4_ADDRESS = 'ipv4_address'
+ATTR_IPV6_ADDRESS = 'ipv6_address'
+ATTR_KERNEL = 'kernel'
+ATTR_MEMORY = 'memory'
+ATTR_REGION = 'region'
+ATTR_VCPUS = 'vcpus'
+
+CONF_DROPLET = 'droplet'
+
+DEFAULT_NAME = 'Droplet'
+DEFAULT_SENSOR_CLASS = 'motion'
+DEPENDENCIES = ['digital_ocean']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DROPLET): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Digital Ocean droplet sensor."""
+    digital_ocean = get_component('digital_ocean')
+    name = config.get(CONF_NAME)
+    droplet = config.get(CONF_DROPLET)
+    droplet_id = digital_ocean.DIGITAL_OCEAN.get_droplet_id(droplet)
+
+    add_devices([DigitalOceanBinarySensor(
+        digital_ocean.DIGITAL_OCEAN, name, droplet_id)])
+
+
+class DigitalOceanBinarySensor(BinarySensorDevice):
+    """Representation of a Digital Ocean droplet sensor."""
+
+    def __init__(self, do, name, droplet_id):
+        """Initialize a new Digital Ocean sensor."""
+        self._name = name
+        self._digital_ocean = do
+        self._droplet_id = droplet_id
+        self._state = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
```

Comment:
I would go for `self.data.name` and _maybe_ allow overriding a name from the config. Even if the default name is too long - it is the name that it is known under.